synapse/handlers/message.py

```diff
@@ -320,6 +321,25 @@ def create_event(self, requester, event_dict, token_id=None, txn_id=None,
                         target, e
                     )
 
+        if builder.type == EventTypes.Aliases:
+            if "aliases" in builder.content:
+                for alias in builder.content["aliases"]:
+                    if len(alias) > MAX_ALIAS_LENGTH:
+                        raise SynapseError(
+                            400,
+                            ("Can't create aliases longer than"
+                             " %s characters" % MAX_ALIAS_LENGTH),
+                            Codes.INVALID_PARAM,
+                        )
+
+            if self.require_membership_for_aliases:
+                rooms_for_user = yield self.store.get_rooms_for_user(builder.sender)
```

Comment:
I think this needs to be based on the actual state of the room at the point in the dag that the event is created, which is part of `context` (below).

Ideally, we'd move it somewhere into the `event_auth` code, though I haven't looked at how tractable it is to put new stuff in there for events which are created by local clients without breaking federation. At the very least we should write comments there and here about why it is not being done in the expected place.

Reply:
Right. FTR, @erikjohnston advised me against doing it in `event_auth`, because that part implements the auth algorithm from the spec and wasn't comfortable with adding something that wasn't spec'd there. I'll do that then.